File: api/ZooOnlineStoreApi/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ZooOnlineStoreApi.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public decimal CostPrice { get; set; }
        public int Quantity { get; set; }
        public string Brand { get; set; } = string.Empty;
        public double Rating { get; set; } = 0;
        public bool isPromotion { get; set; } = false;
        public bool isActive { get; set; } = true;
        //связи
        public int CategoryId { get; set; }
        [ForeignKey(nameof(CategoryId))]
        public Category? Category { get; set; }
        public HashSet<PetType>? PetTypes { get; set; }
        public HashSet<ProductImage>? ProductImages { get; set; }
        public HashSet<Feedback>? Feedbacks { get; set; }

        public Product() { }
    }
}

File: admin/src/models/product.js
export class ProductRequest {
  constructor() {
    this.name = '';
    this.description = '';
    this.price = 0;
    this.costPrice = 0;
    this.quantity = 0;
    this.brand = '';
    this.rating = 0;
    this.isPromotion = false;
    this.isActive = true;
    this.categoryId = 0;
    this.petTypeIds = [];
  }
}
///////
export class ProductResponse {
  constructor() {
    this.id = 0;
    this.name = '';
    this.description = '';
    this.price = 0;
    this.costPrice = 0;
    this.quantity = 0;
    this.brand = '';
    this.rating = 0;
    this.isPromotion = false;
    this.isActive = true;
    this.categoryId = 0;
    this.petTypes = [];
    this.productImages = [];
  }
}
///
export class ProductQueryParameters {
  constructor() {
    //пагинация
    this.page = 1;
    this.pageSize = 10;

    //сортировка и фильтры
    this.isActive = null;
    this.isPromotion = null;
    this.categoryId = null;
    this.petTypeId = null;
    this.name = null;
    this.rating = null;
    this.brand = null;
    this.minPrice = null;
    this.maxPrice = null;
  }
}